raiden/network/proxies/token_network.py

```diff
@@ -809,96 +861,178 @@ def _set_total_deposit(
-                participant2=partner,
-                block_identifier=block_identifier,
-                channel_identifier=channel_identifier,
+            has_sufficient_balance = (
+                self.token.balance_of(self.node_address, given_block_identifier)
+                < amount_to_deposit
             )
-            # Check if deposit is being made on a nonexistent channel
-            if channel_state in (ChannelState.NONEXISTENT, ChannelState.REMOVED):
+            if allowance < amount_to_deposit:
                 msg = (
-                    f"Channel between participant {to_checksum_address(self.node_address)} "
-                    f"and {to_checksum_address(partner)} does not exist"
+                    "The allowance is insufficient. Check concurrent deposits "
+                    "for the same token network but different proxies."
+                )
+            elif has_sufficient_balance:
+                msg = "The address doesnt have enough tokens"
+            else:
+                participant_details = self.detail_participants(
+                    participant1=self.node_address,
+                    participant2=partner,
+                    block_identifier=failed_at_blockhash,
+                    channel_identifier=channel_identifier,
+                )
+                channel_state = self._get_channel_state(
+                    participant1=self.node_address,
+                    participant2=partner,
+                    block_identifier=failed_at_blockhash,
+                    channel_identifier=channel_identifier,
                 )
-            # Deposit was prohibited because the channel is settled
-            elif channel_state == ChannelState.SETTLED:
-                msg = "Deposit is not possible due to channel being settled"
-            # Deposit was prohibited because the channel is closed
-            elif channel_state == ChannelState.CLOSED:
-                error_type = RaidenRecoverableError
-                msg = "Channel is already closed"
-            elif participant_details.our_details.deposit < total_deposit:
-                msg = "Deposit amount did not increase after deposit transaction"
-
-        return error_type, msg
+                # Check if deposit is being made on a nonexistent channel
+                if channel_state in (ChannelState.NONEXISTENT, ChannelState.REMOVED):
+                    msg = (
+                        f"Channel between participant {to_checksum_address(self.node_address)} "
+                        f"and {to_checksum_address(partner)} does not exist"
+                    )
+                # Deposit was prohibited because the channel is settled
+                elif channel_state == ChannelState.SETTLED:
+                    msg = "Deposit is not possible due to channel being settled"
+                # Deposit was prohibited because the channel is closed
+                elif channel_state == ChannelState.CLOSED:
```

Comment:
SETTLED and REMOVED are also recoverable.

Reply:
done